src/PySpectrometer2.py

```diff
@@ -208,330 +252,289 @@ def snapshot(savedata):
-	w1 = img.shape[1]
-	w2 = img_resized.shape[1]
-	xoff = round((w2-w1)/2)
-	# center the background img in the newly sized background
-	img_resized[:, xoff:xoff+w1,:] = img
-	messages = img_resized
-
-	#blank image for Graph
-	graph = np.zeros([320,frameWidth,3],dtype=np.uint8)
-	graph.fill(255) #fill white
-
-	#Display a graticule calibrated with cal data
-	textoffset = 12
-	#vertial lines every whole 10nm
-	for position in tens:
-		cv2.line(graph,(position,15),(position,320),(200,200,200),1)
-
-	#vertical lines every whole 50nm
-	for positiondata in fifties:
-		cv2.line(graph,(positiondata[0],15),(positiondata[0],320),(0,0,0),1)
-		cv2.putText(graph,str(positiondata[1])+'nm',(positiondata[0]-textoffset,12),font,0.4,(0,0,0),1, cv2.LINE_AA)
-
-	#horizontal lines
-	for i in range (320):
-		if i>=64:
-			if i%64==0: #suppress the first line then draw the rest...
-				cv2.line(graph,(0,i),(frameWidth,i),(100,100,100),1)
-	
-	#Now process the intensity data and display it
-	#intensity = []
-	for i in range(cols):
-		#data = bwimage[halfway,i] #pull the pixel data from the halfway mark	
-		#print(type(data)) #numpy.uint8
-		#average the data of 3 rows of pixels:
-		dataminus1 = bwimage[halfway-1,i]
-		datazero = bwimage[halfway,i] #pull the pixel data from the halfway mark
-		dataplus1 = bwimage[halfway+1,i]
-		data = (int(dataminus1)+int(datazero)+int(dataplus1))/3		
-		data = np.uint8(data)
-				
+if not use_picamera:
+	# triggers starting the cap. doing this here to avoid memory/time this takes
+	# from skewing measurements below
+	cap.isOpened()
+
+def runall():
+	global graticuleData, tens, fifties, msg1, saveMsg, waterfall, wavelengthData
+	global caldata, calmsg1, calmsg2, calmsg3, savpoly, mindist, thresh
+	global calibrate, clickArray, cursorX, cursorY, picamGain, spectrum_vertical, waterfall_vertical
+	global graph, graph_base, wavelength_data_rgbs, raw_intensity, picam2, cap
+
+	holdpeaks = False
+	measure = False # show cursor measurements
+	recPixels = False # measure pixels and record clicks
+
+	while (True if use_picamera else cap.isOpened()):
+		# Capture frame-by-frame
+		frame = None
+		ret = True
+		if use_picamera:
+			frame = picam2.capture_array()
+		else:
+			ret, frame = cap.read()
+
+		if not ret:
+			break
+
+		x = 0 # origin of the horiz 
+		h = 80 # height of the crop
+		y = int((frameHeight/2)-(h/2)) # origin of the vertical crop
+		w = frameWidth 	#width of the crop
+		cropped = frame[y:y+h, x:x+w]
+		bwimage = cv2.cvtColor(cropped,cv2.COLOR_BGR2GRAY)
+		rows,cols = bwimage.shape
+		halfway =int(rows/2)
+		#show our line on the original image
+		#now a 3px wide region
+		cv2.line(cropped,(0,halfway-2),(frameWidth,halfway-2),(255,255,255),1)
+		cv2.line(cropped,(0,halfway+2),(frameWidth,halfway+2),(255,255,255),1)
+
+		messages = banner_image
+		# reset the graph to the base
+		np.copyto(graph, graph_base)
+
+		num_mean = 3 # average the data from this many rows of pixels
```

Comment:
This means you could possibly make the number of rows that are averaged a passable argument.